watch/plan.go

```diff
@@ -57,6 +57,7 @@ OUTER:
 		if err != nil {
 			// Perform an exponential backoff
 			failures++
+			p.lastIndex = 1
```

Comment:
Thanks for your reply, @slackpad.

The original fix I can easily reproduce in a test setup (watch() something and restart the Consul server in -dev mode), but a index which goes back (but doesn't drop the connection) I can't test. So I've added the lines you suggested, but I can't test it.

I also changed my original fix to reset the index to 0, not to 1.

(if this is fine I'll squash all commits)